internal/controller/handshake_handler.go

```diff
@@ -0,0 +1,80 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/RedHatInsights/platform-receptor-controller/internal/receptor/protocol"
+)
+
+type HandshakeHandler struct {
+	AccountNumber            string
+	Send                     chan<- Message
+	ControlChannel           chan protocol.Message
+	ErrorChannel             chan error
+	Receptor                 *Receptor
+	Dispatcher               IResponseDispatcher
+	ConnectionMgr            *ConnectionManager
+	MessageDispatcherFactory *MessageDispatcherFactory
+}
+
+func (hh HandshakeHandler) HandleMessage(ctx context.Context, m protocol.Message) error {
+	if m.Type() != protocol.HiMessageType {
+		hh.ErrorChannel <- fmt.Errorf("Invalid message type (type: %d): %v", m.Type(), m)
+		return nil
+	}
+
+	hiMessage, ok := m.(*protocol.HiMessage)
+	if !ok {
+		hh.ErrorChannel <- fmt.Errorf("Unable to convert message into HiMessage")
+		return nil
+	}
+
+	log.Printf("**** got hi message!!  %+v", hiMessage)
+
+	responseHiMessage := protocol.HiMessage{Command: "HI", ID: hh.Receptor.NodeID}
+
+	// FIXME:  this can block...add a timeout and a select here???
+	hh.ControlChannel <- &responseHiMessage // FIXME:  Why a pointer here??
+
+	// FIXME:  What if this account number and node id are already registered?
+	//  abort the connection??
```

Comment:
Would it be a problem to just ignore the message rather than kill the connection?

Reply:
I'm concerned that if we allow another connection from the same account number and node id, then it could be possible for us to send the same message to both connections.  I'm particularly concerned about this once we split the service and start reading messages from the kafka queue.
Does that make sense?

Reply:
Yeah just a misunderstanding on my part!